src/tests/ftest/io/dfuse_bash_cmd.py

```diff
@@ -0,0 +1,212 @@
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+import os
+import general_utils
+
+from ClusterShell.NodeSet import NodeSet
+from apricot import TestWithServers
+from command_utils import CommandFailure
+from daos_utils import DaosCommand
+from test_utils_pool import TestPool
+from test_utils_container import TestContainer
+from dfuse_utils import Dfuse
+
+
+class BashCmd(TestWithServers):
+    """Base BashCmd test class.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a BashCmd object."""
+        super(BashCmd, self).__init__(*args, **kwargs)
+        self.dfuse = None
+        self.file_name = None
+        self.dir_name = None
+        self.pool_count = None
+        self.cont_count = None
+
+    def setUp(self):
+        """Set up each test case."""
+        # Start the servers and agents
+        super(BashCmd, self).setUp()
+
+        # Get the parameters for BashCmd
+        self.dir_name = self.params.get("dirname", '/run/bashcmd/*')
+        self.file_name1 = self.params.get("filename1", '/run/bashcmd/*')
+        self.file_name2 = self.params.get("filename2", '/run/bashcmd/*')
+        self.dd_count = self.params.get("dd_count", '/run/bashcmd/*')
+        self.dd_blocksize = self.params.get("dd_blocksize", '/run/bashcmd/*')
+
+    def tearDown(self):
+        """Tear down each test case."""
+        try:
+            if self.dfuse:
+                self.dfuse.stop()
+        finally:
+            # Stop the servers and agents
+            super(BashCmd, self).tearDown()
+
+    def create_pool(self):
+        """Create a TestPool object to use with ior."""
+        # Get the pool params
+        self.pool = TestPool(
+            self.context, dmg_command=self.get_dmg_command())
+        self.pool.get_params(self)
+
+        # Create a pool
+        self.pool.create()
+
+    def create_cont(self):
+        """Create a TestContainer object to be used to create container."""
+        # Get container params
+        self.container = TestContainer(
+            self.pool, daos_command=DaosCommand(self.bin))
+        self.container.get_params(self)
+
+        # create container
+        self.container.create()
+
+    def start_dfuse(self, count):
+        """Create a DfuseCommand object to start dfuse.
+
+           Args:
+             count(int): container index
+        """
+
+        # Get Dfuse params
+        self.dfuse = Dfuse(self.hostlist_clients[:-1], self.tmp)
```

Comment:
I think this should probably just be self.hostlist_clients.  There have been a number of bugs with how hostlist_clients us used, but using [:-1] is a workaround for the bug so should not be used.